### PR TITLE
Improve Routing and working login with basic profile view

### PR DESCRIPTION
**Pull request by xkb**
@derekdowling had to comment out of some of the post and subscription logic since that data isn't available yet. Other than that,  it works like a charm.

### CODE REVIEW

File: app/scripts/components/author.jsx [hunk 1]
```diff
@@ -10,42 +10,42 @@ import AuthorActions from '../actions/author';
 import ContentViewer from './contentviewer';
 import ContentCreator from './contentcreator';
 import Follow from './follow';
+import Spinner from './spinner';
```
Comment (derekdowling) [inline app/scripts/components/author.jsx:13]: https://www.youtube.com/watch?v=Zfha49oJoDk

File: app/scripts/components/author.jsx [hunk 2]
```diff
@@ -10,42 +10,42 @@ import AuthorActions from '../actions/author';
...
+  //   willTransitionFrom: function () {
+  //     AuthorActions.unbindAuthorListener();
+  //   }
+  // },
 
   getInitialState: function() {
     return {
-      displayAuthor: {},
-      authorPosts: []
+      displayAuthor: null,
```
Comment (derekdowling) [inline app/scripts/components/author.jsx:36]: Do we want null or undefined? I ask to make sure I'm consistent.

Comment (xkb) [inline app/scripts/components/author.jsx:36]: Lets go with null. Undefined infers non existence. My bad haha

Comment (derekdowling) [inline app/scripts/components/author.jsx:36]: I think you can do != null to check undefined as well for checking undefined.

File: app/scripts/components/login.jsx [hunk 3]
```diff
@@ -15,7 +16,7 @@ export default React.createClass({
     willTransitionTo: function (transition, params) {
       // Using the author store is a hack, but until
       // https://github.com/rackt/react-router/pull/590 is merged/closed
-      if (!_.isEmpty(AuthorStore.currentAuthor)) {
+      if (AuthorStore.isLoggedIn()) {
```
Comment (derekdowling) [inline app/scripts/components/login.jsx:19]: Good call. This was bound to turn into a bug especially with no testing.

File: app/scripts/components/login.jsx [hunk 4]
```diff
@@ -28,14 +29,20 @@ export default React.createClass({
     };
   },
 
-  usernameChange: function(evt) {
-    this.setState({username: evt.target.value});
-  },
+  shouldComponentUpdate: function(nextProps) {
```
Comment (derekdowling) [inline app/scripts/components/login.jsx:32]: So, anytime the currentAuthor is updated we get shunted to the timeline?

File: app/scripts/components/author.jsx (hunk 2 above)
Comment (xkb) [inline app/scripts/components/author.jsx:36]: I have nightmares about how `undefined != null` is true.

File: app/scripts/objects/author.js [hunk 5]
```diff
@@ -2,33 +2,43 @@ import _ from 'lodash';
 
 export default class {
 
-  constructor(data, subscriptionStore) {
+  constructor(token, author) {
```
Comment (derekdowling) [inline app/scripts/objects/author.js]: I'd change this to es6 style:

```
constructor(author, token = null)
```

File: app/scripts/components/login.jsx (hunk 4 above)
Comment (xkb) [inline app/scripts/components/login.jsx:32]: This component will only receive the update if it's render so no. You will only be redirect when you're here.

File: app/scripts/objects/author.js [hunk 6]
```diff
@@ -2,33 +2,43 @@ import _ from 'lodash';
...
-    this.subscriptionStore = subscriptionStore;
+    this.token           = token ? token : null;
+    this.id              = author.id;
+    this.bio             = author.bio;
+    this.url             = author.url;
+    this.image           = author.image;
+    this.host            = author.host;
+    this.email           = author.email;
+    this.first_name      = author.first_name;
+    this.last_name       = author.last_name;
```
Comment (derekdowling) [inline app/scripts/objects/author.js:19]: Looks like we should be dealing with just displayName based on the spec now? Can probably do that in a different PR though.

Comment (derekdowling) [inline app/scripts/objects/author.js:19]: NVM, I see it below.

File: app/scripts/objects/author.js [hunk 7]
```diff
@@ -2,33 +2,43 @@ import _ from 'lodash';
...
+    //   var subscriptions = this.subscriptionStore.get(authorId);
+    //   if (_.isUndefined(subscriptions)) {
+    //     this.subscriptionStore.set(authorId, [this.id]);
+    //   } else {
+    //     subscriptions.push(this.id);
+    //   }
+    // }
   }
 
+  getName() {
```
Comment (derekdowling) [inline app/scripts/objects/author.js:39]: Ahh. Perhaps, we should default to username here and then fallback to putting one together?

Comment (xkb) [inline app/scripts/objects/author.js:39]: Username is the `displayname` property and it's independent of this. This is really optional and causes no harm when don't have either fields set.

**Comment by derekdowling** [comment]
Okay, after the author constructor gets updated, :+1:

**Comment by xkb** [comment]
Done.
